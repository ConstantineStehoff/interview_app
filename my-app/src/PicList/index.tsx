import React, { useState, useEffect } from 'react';
import {Spin, List, Card, Layout} from 'antd';
import PicModel from './models/PicModel';
import PicListService from './services/picListService';
import { URL } from '../common/constants';

const picListService = new PicListService();

interface PicListProps {}

const PicList: React.FC<PicListProps> = () => {
    const { Header, Content, Footer } = Layout;
    const [loading, setLoading]  = useState<Boolean>(true);
    const [pics, setPics] = useState<PicModel[]>([]);
    useEffect(() => {
        const getData = async () => { 
            const response:PicModel[] = await picListService.getData(URL);
            setPics(response);
            setLoading(false);
        }
        getData();
    }, []);

    return (
        <Layout>
            {loading && (<Spin size="large" /> )}
            {pics.length > 0 && (
                <>
                    <Header className="site-layout-bg"><h1>List of pictures</h1></Header>
                    <Content className="container">
                        <List 
                            itemLayout="vertical"
                            size="default"
                            pagination={{
                                defaultCurrent: 10
                            }}
                            dataSource={pics}
                            renderItem={item => (
                                <List.Item
                                    key={item.id}
                                >
                                    <div className="job-listing-card">
                                        <Card title={item.title}>
                                            <a href={item.url}>
                                                <img
                                                    width={150}
                                                    alt={item.title}
                                                    src={item.thumbnailUrl}
                                                />
                                            </a>
                                        </Card>
                                    </div>
                                </List.Item>
                            )}
                        />
                    </Content>
                    <Footer><span>Footer 2020</span></Footer>
                </>
            )}
        </Layout>
    )
}

export default PicList;