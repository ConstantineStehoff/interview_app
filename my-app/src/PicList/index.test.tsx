import React from 'react';
import { shallow } from 'enzyme';
import PicList from './index';
test('renders the component', () => {
  const component = shallow(<PicList />);
  expect(component).toMatchSnapshot();
});