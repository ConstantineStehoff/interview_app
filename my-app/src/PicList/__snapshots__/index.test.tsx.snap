// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Layout>
  <Spin
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</Layout>
`;
